src/main/java/au/org/democracydevelopers/raireservice/persistence/entity/GenerateAssertionsSummary.java

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2024 Democracy Developers
+
+The Raire Service is designed to connect colorado-rla and its associated database to
+the raire assertion generation engine (https://github.com/DemocracyDevelopers/raire-java).
+
+This file is part of raire-service.
+
+raire-service is free software: you can redistribute it and/or modify it under the terms
+of the GNU Affero General Public License as published by the Free Software Foundation, either
+version 3 of the License, or (at your option) any later version.
+
+raire-service is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with
+raire-service. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.raireservice.persistence.entity;
+
+import au.org.democracydevelopers.raireservice.service.RaireServiceException;
+import jakarta.persistence.*;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+import static au.org.democracydevelopers.raireservice.service.RaireServiceException.RaireErrorCode.*;
+
+/**
+ * RAIRE (raire-java) generates a set of assertions for a given IRV contest, but it also returns
+ * the winner and (possibly) an informative error. These need to be persisted so that IRV reports
+ * can access the information. This class simply stores the outcome of a call to the
+ * GenerateAssertionsService, one row per contest.
+ */
+@Entity
+@Table(name = "generate_assertions_summary")
+public class GenerateAssertionsSummary {
+
+  private static final Logger logger = LoggerFactory.getLogger(GenerateAssertionsSummary.class);
+
+  /**
+   * Default winner to be used in the case where winner is unknown.
+   */
+  public static final String UNKNOWN_WINNER = "Unknown";
+
+  /**
+   * ID.
+   */
+  @Id
+  @Column(updatable = false, nullable = false)
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+
+  /**
+   * Version. Used for optimistic locking.
+   */
+  @Version
+  @Column(name = "version", updatable = false, nullable = false)
+  private long version;
+
+  /**
+   * Name of the contest.
+   */
+  @Column(name = "contest_name", unique = true, updatable = false, nullable = false)
+  private String contestName = "";
+
+  /**
```

Comment:
Do we mean to make the following attributes public given there are methods for updating them?

Reply:
They have to be read by the getAssertions services - it's really another example of where 'public final' is the right thing, but I don't think final is compatible with the database. You're right, though, they shouldn't be publicly writeable - I've now changed them to private with getters.